modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp

```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
-
-				<div class="<%= publicLayoutSetPrototypeLinkEnabled ? "" : "hide" %>" id="<portlet:namespace />publicLayoutSetPrototypeMergeAlert">
-
-					<%
-					request.setAttribute("edit_layout_set_prototype.jsp-groupId", String.valueOf(siteGoup.getGroupId()));
-					request.setAttribute("edit_layout_set_prototype.jsp-layoutSet", publicLayoutSet);
-					request.setAttribute("edit_layout_set_prototype.jsp-layoutSetPrototype", publicLayoutSetPrototype);
-					request.setAttribute("edit_layout_set_prototype.jsp-redirect", currentURL);
-					%>
-
-					<liferay-util:include page="/layout_set_merge_alert.jsp" servletContext="<%= application %>" />
-				</div>
-			</c:when>
-			<c:when test="<%= publicLayoutSetPrototype != null %>">
-				<liferay-ui:message arguments="<%= HtmlUtil.escape(publicLayoutSetPrototype.getName(locale)) %>" key="these-pages-are-linked-to-site-template-x" translateArguments="<%= false %>" />
-
-				<aui:input name="publicLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
-			</c:when>
-		</c:choose>
-	</c:otherwise>
-</c:choose>
-
-<h4 class="text-default"><liferay-ui:message key="private-pages" /></h4>
 
-<c:choose>
-	<c:when test="<%= (privateLayoutSetPrototype == null) && (siteGoup.getPrivateLayoutsPageCount() == 0) && !layoutSetPrototypes.isEmpty() %>">
-		<c:if test="<%= disableLayoutSetPrototypeInput %>">
-			<div class="alert alert-info">
-				<liferay-ui:message key="you-cannot-apply-a-site-template-because-you-modified-the-display-settings-of-this-site" />
-			</div>
-		</c:if>
-
-		<aui:select disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="site-templates-with-an-incompatible-application-adapter-are-disabled" label="site-template" name="privateLayoutSetPrototypeId">
-			<aui:option label="none" selected="<%= true %>" value="" />
+			<c:choose>
+				<c:when test="<%= (publicLayoutSetPrototype != null) && !siteGoup.isStaged() && hasUnlinkLayoutSetPrototypePermission %>">
+					<c:if test="<%= disableLayoutSetPrototypeInput %>">
+						<div class="alert alert-info">
+							<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+						</div>
+					</c:if>
 
-			<%
-			for (LayoutSetPrototype curLayoutSetPrototype : layoutSetPrototypes) {
-				UnicodeProperties settingsProperties = curLayoutSetPrototype.getSettingsProperties();
-			%>
+					<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(publicLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="publicLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: